### ISSUE TITLE
IContentItemsContainer is not iterable

### ISSUE THREAD

**Issue by drasive** [Kentico/kontent-delivery-sdk-js#267]
### Motivation

IContentItemsContainer is not iterable. This means ContentItem classes with properties like
`public someArrayProperty: Elements.LinkedItemsElement<ContentItem>;` cause GraphQL errors:
`Expected Iterable, but did not find one for field [xxx].someArrayProperty`.

This worked in kentico-cloud-delivery v5.7.2 because the linked items were arrays (https://github.com/Kentico/kontent-delivery-sdk-js/blob/master/UPGRADE.md#linked-items-in-listing-response).

GraphQL type definiton extract: `someArrayProperty: [ContentItem]`
The ContentItem does not contain any custom property resolvers besides the autogenerated underscore to camelCase conversion.

### Proposed solution

Make IContentItemsContainer iterable or provide an example of how to make it work with GraphQL 

Feel free to correct me if some of my assumptions are incorrect or if there is a way to work around this.

### Additional context

Using @kentico/kontent-delivery v8.0.0
Using @kentico/kontent-schema-generator-graphql 2.0.0-rebranding.2
Using @kentico/kontent-model-generator v3.0.0 (ContentItem class generation)
Using graphql v14.3.0 and apollo v2.11.1

**Comment by drasive** [comment]
This might be related to #241

**Comment by Enngage** [comment]
Hey, 

Yeah, the `IContentItemsContainer` is not iterable as I didn't want to convert it to array myself, but keep the structure of what Delivery API returns. There is a fairly easy extension method that can convert it to back to array which I wanted to implement in the issue you linked. 

I'm planning on adding just the method, but not additional array property. I'm assuming this will be ok for you?

**Comment by Enngage** [comment]
I've added the helper method, but its actually very easy to make an array of the linked items object. It's gonna be released with next version. 

It looks like this: 

```
getLinkedItemsAsArray(): IContentItem[] {
   return Object.values(this.items);
}
```

**Comment by drasive** [comment]
Adding the method is useful but doesn't solve the GraphQL issue as the object itself needs to be iterable.  

### Issue
The GraphQL schema generated by @kentico/kontent-schema-generator-graphql 2.0.0-rebranding.2:
```
type SomeContentType implements ContentItem {
    someLinkedProperty: [ContentItem] // Iterable type
    [...]
}
```

Executing the following GraphQL query will lead to an error:
```
{
  itemsBy[...] { // Irrelevant
    ... on SomeContentType {
      someLinkedProperty {
        system { id }
      }
    }
  }
}

=> Expected Iterable, but did not find one for field SomeContentType.someLinkedProperty
```
So the generated GraphQL schema and what is actually returned from @kentico/kontent-delivery v8.0.0 doesn't match.

### GraphQL Schema Workaround
The correct GraphQL schema for what is actually returned by the Kentico Delivery package would be:
```
type IContentItemsContainer {
  value: [ContentItem] // Iterable here
}

type SomeContentType implements ContentItem {
    someLinkedProperty: IContentItemsContainer // Not an iterable type
    [...]
}
```

This enables querying the linked items like this:
```
{
  itemsBy[...] { // Irrelevant
    ... on SomeContentType {
      someLinkedProperty {
        value { // Additional navigation step
          system { id }
        }
      }
    }
  }
}
```

The problem here is the additional navigation step, requiring changes to existing queries. This may be solved somehow.  
I still think this is an issue that should be solved by Kentico itself and not need a workaround.

**Comment by Enngage** [comment]
This is probably more related to the `@kentico/kontent-schema-generator-graphql` rather than this sdk. It would be possible to enhance the object by the schema generator plugin. I'm adding @Simply007 as he is the author for gatsby plugin. Is this something you want to implement?

**Comment by Simply007** [comment]
@Enngage  is right - I have incorrectly set linked items element in the generator:
* https://github.com/Kentico/kontent-schema-generator-graphql/issues/18

**Comment by Simply007** [comment]
And one more fix - should be OK in 2.1.1 ([placed incorrect field name](https://github.com/Kentico/kontent-schema-generator-graphql/commit/ab4e593692b6adc96dd738d4f1039146667a36ef)) of graphql schema generator
